File: src/com/lysdev/transperthcached/silverrails/NearbyTransitStop.java
package com.lysdev.transperthcached.silverrails;


public class NearbyTransitStop extends TransitStop {
    public NearbyTransitStop(
            int distance,

            double latitude, double longitude,
            String description,
            int code
            ) {
        super(latitude, longitude, description, code);
        this.distance = distance;
    }

    public String toString() {
        return String.format(
            "%d metres to %s",
            getDistance().intValue(),
            getDescription()
        );
    }


    private Integer distance;

    public Integer getDistance() {
        return this.distance;
    }
}

File: src/com/lysdev/transperthcached/silverrails/NearbyTransitStop.scala
package com.lysdev.transperthcached.silverrails

class NearbyTransitStop(
            distance: Int,

            latitude: Double, longitude: Double,
            description: String,
            code: Int
            ) extends TransitStop(latitude, longitude, description, code) {

    override
    def toString() = f"$getDistance.intValue metres to $getDescription"
    def getDistance() = this.distance
}
